src/hooks/useShowNotFoundPageInIOUStep.ts

```diff
@@ -0,0 +1,53 @@
+import {useMemo} from 'react';
+import type {OnyxEntry} from 'react-native-onyx';
+import {isMoneyRequestAction} from '@libs/ReportActionsUtils';
+import {canEditMoneyRequest} from '@libs/ReportUtils';
+import {areRequiredFieldsEmpty} from '@libs/TransactionUtils';
+import type {IOUAction, IOUType} from '@src/CONST';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {OnyxInputOrEntry, Report, Transaction} from '@src/types/onyx';
+import useOnyx from './useOnyx';
+
+// eslint-disable-next-line rulesdir/no-negated-variables
+const useShowNotFoundPageInIOUStep = (action: IOUAction, iouType: IOUType, reportActionID: string | undefined, report: OnyxInputOrEntry<Report>, transaction: OnyxEntry<Transaction>) => {
```

Comment:
Can you please add a plain english doc here listing out all the cases and flows where we should the not found page? what are the conditions